src/utils/datatool/SceneLoader.cpp

```diff
@@ -83,32 +81,36 @@ MeshData SceneLoader::load(const AssetInfo& info) {
 
     scene = Importer.ReadFile(info.filepath.c_str(), assimpFlags);
 
-    const quatf alignSceneToEspGravity =
-        quatf::FromTwoVectors(info.frame.gravity(), esp::geo::ESP_GRAVITY);
+    const Mn::Quaternion alignSceneToEspGravity = core::quatRotFromTwoVectors(
+        info.frame.gravity(), esp::geo::ESP_GRAVITY);
 
     // Iterate through all meshes in the file and extract the vertex components
     for (uint32_t m = 0, indexBase = 0; m < scene->mNumMeshes; ++m) {
       const aiMesh& assimpMesh = *scene->mMeshes[m];
       for (uint32_t v = 0; v < assimpMesh.mNumVertices; ++v) {
-        // Use Eigen::Map to convert ASSIMP vectors to eigen vectors
-        const Eigen::Map<const vec3f> xyz_scene(&assimpMesh.mVertices[v].x);
-        const vec3f xyz_esp = alignSceneToEspGravity * xyz_scene;
+        const Mn::Vector3 xyz_scene =
+            Mn::Vector3::from(&assimpMesh.mVertices[v].x);
+        const Mn::Vector3 xyz_esp =
+            alignSceneToEspGravity.transformVectorNormalized(xyz_scene);
         mesh.vbo.push_back(xyz_esp);
 
         if (assimpMesh.mNormals) {
```

Comment:
Um, there's a whole `AssimpImporter` for this kind of thing. Can't the `datatool` be just deleted or something?

A cleaner and _safer_ way (bounds checking, type compatiblity checking...) instead of `Mn::Vector3::from()` etc would be to take the whole `mNormals` array, make an array view and cast it:

```cpp
Cr::Containers::ArrayView<const Mn::Vector3> normals = 
    Cr::Containers::arrayCast<const Mn::Vector3>(Cr::Containers::arrayView(assimpMesh.mNormals, assimpMesh.mNumVertices));
```

And similar for positions etc.

Reply:
Yeah, I know, I just don't want to make that decision unilaterally. @aclegg3 Thoughts?